code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,464 @@
+		/obj/item/reagent_containers/hypospray/CMO,
+		/obj/item/clothing/accessory/medal/gold/captain,
+		/obj/item/clothing/gloves/krav_maga,
+		/obj/item/storage/internal,
+		/obj/item/nullrod
+	)
+	// These items will NOT be preserved
+	var/list/do_not_preserve_items = list (
+		/obj/item/device/mmi/posibrain
+	)
+
+/obj/machinery/cryopod/Initialize()
+	..()
+	update_icon()
+	find_control_computer()
+
+/obj/machinery/cryopod/proc/find_control_computer(urgent=0)
+	for(var/obj/machinery/computer/cryopod/C in area_contents(get_area(src)))
+		control_computer = C
+		break
+
+	// Don't send messages unless we *need* the computer, and less than five minutes have passed since last time we messaged
+	if(!control_computer && urgent && last_no_computer_message + 5*60*10 < world.time)
+		log_admin("Cryopod in [get_area(src)] could not find control computer!")
+		message_admins("Cryopod in [get_area(src)] could not find control computer!")
+		last_no_computer_message = world.time
+
+	return control_computer != null
+
+/obj/machinery/cryopod/close_machine(mob/user)
+	if((isnull(user) || istype(user)) && state_open && !panel_open)
+		..(user)
+		var/mob/living/mob_occupant = occupant
+		if(mob_occupant && mob_occupant.stat != DEAD)
+			to_chat(occupant, "<span class='notice'><b>You feel cool air surround you. You go numb as your senses turn inward.</b></span>")
+		if(mob_occupant.client)//if they're logged in
+			despawn_world_time = world.time + (time_till_despawn * 0.1)
+		else
+			despawn_world_time = world.time + time_till_despawn
+	icon_state = "cryopod"
+
+/obj/machinery/cryopod/open_machine()
+	..()
+	icon_state = "cryopod-open"
+	//TODO?
+
+/obj/machinery/crypod/container_resist(mob/living/user)
+	visible_message("<span class='notice'>[occupant] emerges from [src]!</span>",
+		"<span class='notice'>You climb out of [src]!</span>")
+	open_machine()
+
+/obj/machinery/cryopod/relaymove(mob/user)
+	container_resist(user)
+
+/obj/machinery/cryopod/process()
+	if(!occupant)
+		return
+
+	var/mob/living/mob_occupant = occupant
+	if(mob_occupant)
+		// Eject dead people
+		if(mob_occupant.stat == DEAD)
+			open_machine()
+
+		if(!(world.time > despawn_world_time + 100))//+ 10 seconds
+			return
+
+		if(!mob_occupant.client && mob_occupant.stat < 2) //Occupant is living and has no client.
+			if(!control_computer)
+				if(!find_control_computer(urgent = 1))
+					return
+
+			despawn_occupant()
+
+// This function can not be undone; do not call this unless you are sure
+/obj/machinery/cryopod/proc/despawn_occupant()
+	var/mob/living/mob_occupant = occupant
+	for(var/obj/item/W in mob_occupant.GetAllContents())
+		W.forceMove(src)
+		for(var/T in preserve_items)
+			if(!istype(W, T))
+				if(control_computer && control_computer.allow_items)
+					control_computer.frozen_items += W
+					W.loc = null
+				else
+					W.forceMove(loc)
+			else
+				qdel(W)
+	if(istype(SSticker.mode, /datum/game_mode/cult))//thank
+		if("sacrifice" in SSticker.mode.cult_objectives)
```

Comment:
Will this trigger for all cult forms? Clock?

Reply:
Clock cult doesn't get any objectives. Just grow and "summon".